smithy-jmespath/src/main/java/software/amazon/smithy/jmespath/Lexer.java

```diff
@@ -5,17 +5,17 @@
 package software.amazon.smithy.jmespath;
 
 import java.util.ArrayList;
-import java.util.LinkedHashMap;
 import java.util.List;
-import java.util.Map;
 import java.util.Objects;
 import java.util.function.Predicate;
 import software.amazon.smithy.jmespath.ast.LiteralExpression;
+import software.amazon.smithy.jmespath.evaluation.JmespathRuntime;
 
-final class Lexer {
+final class Lexer<T> {
```

Comment:
The type param is weird, because it returns `T` for both, values and arrays. Probably a bound will be better but for that some kind of marker interface would be needed.